dask/array/tests/test_xarray.py

```diff
@@ -22,3 +22,15 @@ def test_asanyarray():
     y = da.asanyarray(xr.DataArray([1, 2, 3.0]))
     assert isinstance(y, da.Array)
     assert_eq(y, y)
+
+
+def test_asarray_recognises_xarray_data():
```

Comment:
Could you add a comment here which points out that this test ensures we support users who are using the intersphinx workaround in https://github.com/pydata/xarray/issues/4279